### PR TITLE
server: Actively prevent regnet network discovery

### PR DESCRIPTION
**Pull request by chappjc**
This treats regnet like simnet with respect to node discovery and outbound connections.

Without this change, using regnet without `--connect`/`ConnectPeers` results in peer connection errors being logged ad nauseum.  e.g. tons of logs like `Failed to connect to reqid 78: no valid connect address`, eventually giving up.,
The workaround is to always use `ConnectPeers` on regnet, but peer discovery is really best disabled for regnet as well as simnet.

A similar change was made in dcrd in https://github.com/decred/dcrd/commit/f5fdd95eb578a485b414331836f9731fcd0f2756

Would like to test a bit more to be sure nothing unexpected happens on our regnet harness, so in draft for the moment.

### CODE REVIEW

File: neutrino.go [hunk 1]
```diff
@@ -320,7 +320,8 @@ func (sp *ServerPeer) OnAddr(_ *peer.Peer, msg *wire.MsgAddr) {
 	// helps prevent the network from becoming another public test network
 	// since it will not be able to learn about other peers that have not
 	// specifically been provided.
-	if sp.server.chainParams.Net == chaincfg.SimNetParams.Net {
+	if sp.server.chainParams.Net == chaincfg.SimNetParams.Net ||
```
Comment (Roasbeef) [inline neutrino.go:323]: How about we spin these repeated conditionals out into something like `isDevNetwork`? Applies also in the diff for the other file.

**Review by Roasbeef — CHANGES_REQUESTED** [review]

File: neutrino.go (hunk 1 above)
Comment (chappjc) [reply neutrino.go]: Made a `isDevNetwork` helper taking a `wire.BitcoinNet`.  Seemed cleanest.

File: blockmanager.go [hunk 2]
```diff
@@ -2680,9 +2680,7 @@ func (b *blockManager) handleHeadersMsg(hmsg *headersMsg) {
 
 	// If not current, request the next batch of headers starting from the
 	// latest known header and ending with the next checkpoint.
-	if b.cfg.ChainParams.Net == chaincfg.SimNetParams.Net ||
-		b.cfg.ChainParams.Net == chaincfg.RegressionNetParams.Net ||
-		!b.BlockHeadersSynced() {
+	if isDevNetwork(b.cfg.ChainParams.Net) || !b.BlockHeadersSynced() {
```
Comment (ellemouton) [inline blockmanager.go:2683]: nit: I dont think this one is necessary. Looks like this is only done for unit testing purposes.

File: neutrino.go [hunk 3]
```diff
@@ -747,8 +753,7 @@ func NewChainService(cfg Config) (*ChainService, error) {
 	// peers and actively avoid advertising and connecting to discovered
 	// peers in order to prevent it from becoming a public test network.
 	var newAddressFunc func() (net.Addr, error)
-	if s.chainParams.Net != chaincfg.SimNetParams.Net &&
-		s.chainParams.Net != chaincfg.RegressionNetParams.Net {
+	if !isDevNetwork(s.chainParams.Net) {
```
Comment (ellemouton) [inline neutrino.go:756]: nit: can you update the comment above to indicate that we are no longer just switching on simnet here.

File: neutrino.go [hunk 4]
```diff
@@ -1265,8 +1270,7 @@ func (s *ChainService) handleAddPeerMsg(state *peerState, sp *ServerPeer) bool {
 	// the simulation test network since it is only intended to connect to
 	// specified peers and actively avoids advertising and connecting to
 	// discovered peers.
-	if s.chainParams.Net != chaincfg.SimNetParams.Net &&
-		s.chainParams.Net != chaincfg.RegressionNetParams.Net {
+	if !isDevNetwork(s.chainParams.Net) {
```
Comment (ellemouton) [inline neutrino.go:1273]: same as above re comment

File: neutrino.go [hunk 5]
```diff
@@ -320,8 +327,7 @@ func (sp *ServerPeer) OnAddr(_ *peer.Peer, msg *wire.MsgAddr) {
 	// helps prevent the network from becoming another public test network
 	// since it will not be able to learn about other peers that have not
 	// specifically been provided.
-	if sp.server.chainParams.Net == chaincfg.SimNetParams.Net ||
-		sp.server.chainParams.Net == chaincfg.RegressionNetParams.Net {
+	if isDevNetwork(sp.server.chainParams.Net) {
```
Comment (ellemouton) [inline neutrino.go:330]: same as below re comment

File: neutrino.go [hunk 6]
```diff
@@ -84,6 +84,13 @@ var (
 	DefaultBlockCacheSize uint64 = 4096 * 10 * 1000 // 40 MB
 )
 
+// isDevNetwork indicates if the chain is a development network, namely simnet
+// or regtest/regnet.
+func isDevNetwork(net wire.BitcoinNet) bool {
+	return net == chaincfg.SimNetParams.Net ||
+		net == chaincfg.RegressionNetParams.Net
+}
+
```
Comment (ellemouton) [inline neutrino.go:93]: This commit can be squashed into the first commit

**Review by ellemouton — COMMENTED** [review]
tACK 🚀  just pending some nits re comments. Also think this should all be squashed into 1 commit. I also think the change in `blockmanager.go` is not needed for this.

File: blockmanager.go (hunk 2 above)
Comment (chappjc) [reply blockmanager.go]: I reverted this one back to just simnet, so no changes on blockmanager.go anymore.

File: neutrino.go (hunk 3 above)
Comment (chappjc) [reply neutrino.go:756]: Done.  Sorry about re-wrapping the comment.  It was starting to look wonky after making the change.

**Review by ellemouton — APPROVED** [review]
LGTM 🚀

**Comment by chappjc** [comment]
Thanks, @lightninglabs-deploy, it's already approved. :stuck_out_tongue_winking_eye:

**Comment by ellemouton** [comment]
@chappjc , it needs 2 approvals :) so just need to re-request review from the other reviewer

**Review by Roasbeef — APPROVED** [review]
LGTM 🦧

**Pull request merged**